mtwitter.js

```diff
@@ -0,0 +1,307 @@
+        }
+      });
+    });
+  });
+}
+
+MTwitter.prototype.get_user_by_id = function(id) {
+    var ts = new Date().getTime();
+    return new Promise((resolve, reject) => {
+        for (var cached of this.tuser_cache) {
+            if (ts - cached.cache_time > TWITTER_PROFILE_INTERVAL_MS) {
+                continue;
+            }
+            if (cached.user != null && cached.user.id_str == id) {
+                resolve(cached.user);
+            }
+        }
+        this.app_twitter.get('users/show', {
+            user_id: id
+        }, (error, user, response) => {
+            if (error) {
+                console.error("get_user_by_id: GET /users/show returned: ", error);
+                reject(error);
+            } else {
+                this.tuser_cache[user.screen_name] = {
+                    cache_time: ts,
+                    user: user
+                };
+                resolve(user);
+            }
+        });
+    });
+}
+
+MTwitter.prototype.get_user = function(name) {
+    console.log("Looking up @" + name);
+    return new Promise((resolve, reject) => {
+        var ts = new Date().getTime();
+        if (this.tuser_cache[name] != undefined) {
+            console.log("Checking cache for @" + name);
+            var cached = this.tuser_cache[name];
+            if (ts - cached.cache_time < TWITTER_PROFILE_INTERVAL_MS) {
+                resolve(cached.user);
+            }
+        }
+
+        console.log("Checking twitter for @" + name);
+        this.app_twitter.get('users/show', {
+            screen_name: name
+        }, (error, user, response) => {
+            if (error) {
+                this.tuser_cache[name] = {
+                    cache_time: ts,
+                    user: null
+                };
+                reject(error);
+            }
+            this.tuser_cache[name] = {
+                cache_time: ts,
+                user: user
+            };
+            resolve(user);
+        });
+    });
+
+}
+
+/*
+  Add a user's timeline to the timeline processor. Tweets will be automatically send to
+  the given room.
+*/
+MTwitter.prototype.add_timeline = function(userid, localroom, remoteroom) {
+    var obj = {
+        "user_id": userid,
+        "local": localroom,
+        "remote": remoteroom
+    };
+    this.timeline_list.push(obj);
+    this.timeline_queue.push(obj);
+}
+
+/*
+  This function will fill the content structure for a new matrix message
+  for a given tweet.
+*/
+MTwitter.prototype.tweet_to_matrix_content = function(tweet, type) {
+    return {
+        "body": tweet.text,
+        "created_at": tweet.created_at,
+        "likes": tweet.favorite_count,
+        "reblogs": tweet.retweet_count,
+        "tweet_id": tweet.id_str,
+        "tags": tweet.entities.hashtags,
+        "msgtype": type
+    }
+}
+
+//Runs every 500ms to help not overflow the room.
+MTwitter.prototype._process_queue = function(){
```

Comment:
It would be a tad clearer to call this something like `_process_head_of_queue` to make it clear you're not fully emptying the queue (which is what I would expect with a "process queue" function).